src/relay/qnn/op/convolution.cc

```diff
@@ -40,6 +40,23 @@ namespace qnn {
 // relay.op.qnn.conv2d
 TVM_REGISTER_NODE_TYPE(QnnConv2DAttrs);
 
+bool QnnConv2DRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
+               const TypeReporter& reporter) {
+  CHECK_EQ(types.size(), 3);
+  const auto* data = types[0].as<TensorTypeNode>();
+  const auto* weight = types[1].as<TensorTypeNode>();
+  const auto* param = attrs.as<QnnConv2DAttrs>();
+  CHECK(data->dtype == Int(8) || data->dtype == UInt(8))
+    << "Expected quantized dense type(int8, uint8) for input but was " <<  data->dtype;
+  CHECK(weight->dtype == Int(8) || weight->dtype == UInt(8))
+    << "Expected quantized dense type(int8, uint8) for weight but was " <<  weight->dtype;
+  CHECK(data->dtype == weight->dtype) << "Weight and kernel dtypes do not match";
```

Comment:
I dont think we need this check. Weight and Input dtypes can be different. I think TFLite new proposal wants the activations to be uint8 and weights to be int8.